src/main/java/io/cdap/plugin/sfmc/connector/MarketingCloudConnector.java

```diff
@@ -0,0 +1,257 @@
+        browseDetailBuilder.addEntity(entity.build());
+      } else {
+        BrowseEntity.Builder entity = (BrowseEntity.builder(String.valueOf(object), String.valueOf(object),
+          ENTITY_TYPE_OBJECTS).canBrowse(false).canSample(true));
+        browseDetailBuilder.addEntity(entity.build());
+      }
+    }
+    return browseDetailBuilder.setTotalCount(SourceObject.values().length).build();
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest) {
+    ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
+    Map<String, String> properties = new HashMap<>();
+    properties.put(ConfigUtil.NAME_USE_CONNECTION, "true");
+    properties.put(ConfigUtil.NAME_CONNECTION, connectorSpecRequest.getConnectionWithMacro());
+    String objectName = connectorSpecRequest.getPath();
+    MarketingCloudClient marketingCloudClient = null;
+    try {
+      marketingCloudClient = MarketingCloudUtil.getInstance(config.getClientId(),
+        config.getClientSecret(), config.getAuthEndpoint(), config.getSoapEndpoint());
+    } catch (ETSdkException e) {
+      throw new RuntimeException(e);
+    }
+    if (objectName != null && !objectName.startsWith(SourceObject.DATA_EXTENSION.toString())) {
+      properties.put(MarketingCloudConstants.PROPERTY_OBJECT_NAME, SourceObject.valueOf(objectName).getValue());
+      try {
+        schema = config.getSchema(SourceObject.valueOf(objectName));
+      } catch (ETSdkException e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      objectName = objectName.substring(objectName.indexOf("/") + 1);
+      properties.put(MarketingCloudConstants.PROPERTY_OBJECT_NAME, SourceObject.DATA_EXTENSION.getValue());
+      properties.put(MarketingCloudConstants.PROPERTY_DATA_EXTENSION_KEY, objectName);
+      try {
+        schema = marketingCloudClient.fetchDataExtensionSchema(objectName).getSchema();
+      } catch (ETSdkException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    properties.put(Constants.Reference.REFERENCE_NAME, ReferenceNames.cleanseReferenceName(objectName));
+    specBuilder.setSchema(schema);
+    return specBuilder.addRelatedPlugin(new PluginSpec(MarketingCloudConstants.PLUGIN_NAME, BatchSource.PLUGIN_TYPE,
+      properties)).addRelatedPlugin(new PluginSpec(MarketingCloudDataExtensionSink.PLUGIN_TYPE,
+      BatchSink.PLUGIN_TYPE, properties)).build();
+  }
+
+  @Override
+  public List<StructuredRecord> sample(ConnectorContext connectorContext, SampleRequest sampleRequest)
+    throws IOException {
+    MarketingCloudClient marketingCloudClient = null;
+    MarketingCloudUtil cloudUtil = new MarketingCloudUtil();
+    try {
+      marketingCloudClient = MarketingCloudUtil.getInstance(config.getClientId(),
+        config.getClientSecret(), config.getAuthEndpoint(), config.getSoapEndpoint());
+    String objectName = sampleRequest.getPath();
```

Comment:
Correct the line indentations in all the files.

Reply:
Fixed the line indentation in all the files. Please have a look.